Update the checked input radio button margin style

hsingyuc opened the pull request:
Fixes #6651 

Update the checked input radio button margin style.

### Screenshots
Before
<img width="378" alt="before" src="https://user-images.githubusercontent.com/56378160/112911900-5a6cc680-90c4-11eb-805f-50618d427512.png">

After
<img width="389" alt="after" src="https://user-images.githubusercontent.com/56378160/112911903-5e98e400-90c4-11eb-8638-e2ba21352b6c.png">


### Detailed test instructions:

- Go to Home
- Click on 'Add my products'
- Select 'Start with a template'
- Click on the input radio button and see that render as expected

<!-- 
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note following the WooCommerce core format using prefixes of Enhancement:, Tweak:, Dev:, Fix:, Performance: to readme.txt under the "unreleased" list at the top of the changelog. If none exists, please add it. Also include PR number.

If changes pertain to a package, also update CHANGELOG.md in the package's folder in a similar manner.--->

mattsherman commented:
@hsingyuc Thanks for the fix -- I'll test and review this today!

mattsherman reviewed (changes requested):
@hsingyuc Thanks for working on a fix for this issue! And sorry for taking so long for us to review this. I have tested the PR and unfortunately there are still layout issues on mobile viewports:

<img width="463" alt="Screen Shot 2021-04-09 at 21 51 52" src="https://user-images.githubusercontent.com/2098816/114254537-fc6e9780-997d-11eb-8bf9-ad03951613a8.png">

Setting `margin: -7px 0 0 -7px` for the smaller viewport fixes the issue.

But, this margin manipulation ultimately feels brittle (I realize that this was already in place and this PR just adjusts things). I wonder if there is a more robust way of changing the colors of the radio control that won't as easily break if the `@wordpress/components` radio component changes? 

If not, I think that the basic fix you implemented (with support for the mobile viewports as well) is acceptable.

Please also rebase against `main` and resolve the conflicts in the `TESTING-INSTRUCTIONS.md` and `readme.txt` files. 

Thanks again for your contribution!

hsingyuc commented:
@mattsherman  I changed the `List` to `RadioControl` to prevent breaking changes from Gutenberg styles. It looks like the `List` styles were partially overwritten.

mattsherman commented on client/task-list/tasks/products/product-template-modal.js:
```diff
@@ -104,36 +98,25 @@ export default function ProductTemplateModal( { onClose } ) {
+							return {
+								label: (
+									<>
+										<div className="woocommerce-list__item-label">
```
It feels "wrong" to use the `woocommerce-list__item-label` class here, as this isn't the intended use of the class. I think it would be better to either create a specific class for this, or even better, use the `Text` component from `@wordpress/components`.

mattsherman commented on client/task-list/tasks/products/product-template-modal.js:
```diff
@@ -104,36 +98,25 @@ export default function ProductTemplateModal( { onClose } ) {
+										<div className="woocommerce-list__item-label">
+											{ item.title }
+										</div>
+										<div className="woocommerce-list__item-subtitle">
```
Same comment as above.

mattsherman commented on client/task-list/tasks/products/product-template-modal.scss:
```diff
@@ -8,24 +8,33 @@ $border-color: $gray-100;
-			}
-			&:focus {
-				box-shadow: none;
+	.components-base-control {
+		margin: 0 -#{$gap-large};
+		.components-base-control__field {
+			.components-radio-control__option {
+				display: flex;
+				text-decoration: none;
+				width: 100%;
+				align-items: center;
+				padding: $gap $gap-large;
+				margin-bottom: 0;
+				box-sizing: border-box;
+				border-bottom: 1px solid $gray-100;
+				&:first-child {
+					border-top: 1px solid $gray-100;
+				}
+				.components-radio-control__input {
+					margin-right: $gap;
+					width: 20px;
+					height: 20px;
+					margin-right: $gap;
+					&:checked::before {
+						box-sizing: border-box;
+					}
+					&:focus {
+						box-shadow: none;
+					}
+				}
```
Why does all of this need to be specified? I would think we would get almost all of what we need just from the core component.

mattsherman reviewed (changes requested):
@hsingyuc I think the idea to use `RadioControl` instead of `List` is great -- that feels like a much better fit.

There are still some styling issues on mobile viewport sizes:

<img width="463" alt="Screen Shot 2021-04-14 at 20 25 50" src="https://user-images.githubusercontent.com/2098816/114796720-9e67f880-9d5f-11eb-9f07-984268951e13.png">

hsingyuc commented:
@mattsherman  Thank you for the review. I ended up using `span` instead of the `Text` component because there was no matching style we can use. Changed the class name to `woocommerce-product-template-modal__list-title`. Do you think that's ok?

Also fixed an issue with the modal on smaller screens

mattsherman commented on client/task-list/tasks/products/product-template-modal.scss:
```diff
@@ -1,43 +1,50 @@
+				&:first-child {
+					border-top: 1px solid $gray-100;
+				}
+				.components-radio-control__input {
```
We need a `flex: none;` property on `.components-radio-control__input`. Otherwise, the radio button will stretch in some scenarios.

iPhone X:

<img width="464" alt="Screen Shot 2021-04-19 at 15 21 55" src="https://user-images.githubusercontent.com/2098816/115292840-83610400-a124-11eb-9dae-a028df9d52cc.png">

mattsherman reviewed (changes requested):
Thanks for your commitment to seeing this PR through, @hsingyuc !

I think there is just one minor issue left to address, as well as fixing the conflicts in the `README.txt` and `TESTING-INSTRUCTIONS.md` files.

We would love to get this fix into the next release. A beta release of it is going to be built tomorrow morning. So, if you are able to get this completed by then we can include it! If not, we could have somebody else make the final changes.

/cc @timmyc

hsingyuc commented:
@mattsherman  Thank you so much for the feedback! The changes have been pushed.

mattsherman reviewed (approved):
Thanks for the final fixes @hsingyuc -- this looks good! Nice work.
